File: backend/internal/constants/proposal_state.go
package constants

// ProposalState describes the internal review lifecycle; accepted never means legal registration.
type ProposalState string

const (
	ProposalDraft     ProposalState = "draft"
	ProposalValidated ProposalState = "validated"
	ProposalSubmitted ProposalState = "submitted"
	ProposalReviewed  ProposalState = "reviewed"
	ProposalRevision  ProposalState = "revision"
	ProposalAccepted  ProposalState = "accepted"
	ProposalRejected  ProposalState = "rejected"
)

var proposalTransitions = map[ProposalState]map[ProposalState]bool{
	ProposalDraft:     {ProposalValidated: true},
	ProposalValidated: {ProposalSubmitted: true},
	ProposalSubmitted: {ProposalReviewed: true},
	ProposalReviewed:  {ProposalAccepted: true, ProposalRejected: true, ProposalRevision: true},
	ProposalRevision:  {ProposalDraft: true},
	ProposalAccepted:  {}, ProposalRejected: {},
}

func (s ProposalState) Valid() bool                     { _, ok := proposalTransitions[s]; return ok }
func CanProposalTransition(from, to ProposalState) bool { return proposalTransitions[from][to] || proposalTransitions[to][from] }

File: frontend/src/types/enums/proposal-state.ts
export const PROPOSAL_STATES = ['draft', 'validated', 'submitted', 'reviewed', 'revision', 'accepted', 'rejected'] as const
export type ProposalState = (typeof PROPOSAL_STATES)[number]
export const proposalStateLabel: Record<ProposalState, string> = { draft: '草拟', validated: '已校验', submitted: '已提交', reviewed: '已复核', revision: '待修订', accepted: '内部采纳', rejected: '已拒绝' }
